xbmc/cores/VideoPlayer/DVDSubtitles/DVDFactorySubtitle.cpp

```diff
@@ -21,132 +21,62 @@
 #include "DVDFactorySubtitle.h"
 
 #include "DVDSubtitleStream.h"
-//#include "DVDSubtitleParserSpu.h"
 #include "DVDSubtitleParserSubrip.h"
 #include "DVDSubtitleParserMicroDVD.h"
 #include "DVDSubtitleParserMPL2.h"
 #include "DVDSubtitleParserSami.h"
 #include "DVDSubtitleParserSSA.h"
 #include "DVDSubtitleParserVplayer.h"
 
+#include <memory>
+
 CDVDSubtitleParser* CDVDFactorySubtitle::CreateParser(std::string& strFile)
 {
   char line[1024];
   int i;
-  CDVDSubtitleParser* pParser = NULL;
 
-  CDVDSubtitleStream* pStream = new CDVDSubtitleStream();
+  std::unique_ptr<CDVDSubtitleStream> pStream(new CDVDSubtitleStream());
   if(!pStream->Open(strFile))
   {
-    delete pStream;
-    return NULL;
+    return nullptr;
   }
 
-  for (int t = 0; !pParser && t < 256; t++)
+  for (int t = 0; t < 256; t++)
   {
     if (pStream->ReadLine(line, sizeof(line)))
     {
       if ((sscanf (line, "{%d}{}", &i)==1) ||
           (sscanf (line, "{%d}{%d}", &i, &i)==2))
       {
-        pParser = new CDVDSubtitleParserMicroDVD(pStream, strFile.c_str());
-        pStream = NULL;
+        return new CDVDSubtitleParserMicroDVD(std::move(pStream), strFile.c_str());
```

Comment:
move does transfer the ownership as well. what do you think is the difference to?

return new CDVDSubtitleParserMicroDVD(pStream.release(), strFile.c_str());